apps/backend/src/auth/StfcUserAuthorization.ts

```diff
@@ -203,9 +203,9 @@ export class StfcUserAuthorization extends UserAuthorization {
 
     // Auto-assign users to instruments.
     // This will happen if the user is an instrument scientist for an STFC facility
-    const stfcRoles: stfcRole[] | null = (
-      await client.getRolesForUser(process.env.EXTERNAL_AUTH_TOKEN, userNumber)
-    )?.return;
+    const stfcRoles: stfcRole[] | null = await (
+      this.userDataSource as StfcUserDataSource

```

Comment:
Is the cast to StfcUserDataSource necessary? It looks like the variable is already typed that way.